### PR TITLE
chore: use eslint-config-ipfs

### PR DESCRIPTION
**Pull request by Gozala**
Demonstrates use shared eslint config `eslint-config-ipfs` by integrating ipfs/aegir#638. Shared config uses new jsdoc plugin enabling use to remove bunch of `eslint-disable-next-line valid-jsdoc` comments. On the flip side it adds bit more requirements for structuring jsdoc comments. Most changes are generated automatically by eslint with `--fix` flag.

This change also intentionally adds `.eslintrc` files to every package with content `{ "extends": "ipfs" }`, so that all the tools that support eslint pick up desired configuration.  `eslint-config-ipfs` package itself isn't added to dependencies, because it gets installed as `aegir` dependency.

### CODE REVIEW

File: packages/interface-ipfs-core/package.json [hunk 1]
```diff
@@ -62,7 +62,7 @@
     "uint8arrays": "^1.1.0"
   },
   "devDependencies": {
-    "aegir": "^26.0.0",
+    "aegir": "git://github.com/ipfs/aegir#eslint-config",
```
Comment (Gozala) [inline packages/interface-ipfs-core/package.json:65]:
```suggestion
    "aegir": "git://github.com/ipfs/aegir#master",
```

File: packages/ipfs-core-utils/package.json [hunk 2]
```diff
@@ -39,7 +39,7 @@
     "uint8arrays": "^1.1.0"
   },
   "devDependencies": {
-    "aegir": "^26.0.0",
+    "aegir": "git://github.com/ipfs/aegir#eslint-config",
```
Comment (Gozala) [inline packages/ipfs-core-utils/package.json:42]:
```suggestion
    "aegir": "git://github.com/ipfs/aegir#master",
```

File: packages/ipfs-http-client/package.json [hunk 3]
```diff
@@ -71,7 +71,7 @@
     "uint8arrays": "^1.1.0"
   },
   "devDependencies": {
-    "aegir": "^26.0.0",
+    "aegir": "git://github.com/ipfs/aegir#eslint-config",
```
Comment (Gozala) [inline packages/ipfs-http-client/package.json:74]:
```suggestion
    "aegir": "git://github.com/ipfs/aegir#master",
```

File: packages/ipfs-message-port-client/package.json [hunk 4]
```diff
@@ -38,7 +38,7 @@
     "cids": "^1.0.0"
   },
   "devDependencies": {
-    "aegir": "^26.0.0",
+    "aegir": "git://github.com/ipfs/aegir#eslint-config",
```
Comment (Gozala) [inline packages/ipfs-message-port-client/package.json:41]:
```suggestion
    "aegir": "git://github.com/ipfs/aegir#master",
```

File: packages/ipfs-message-port-protocol/package.json [hunk 5]
```diff
@@ -36,7 +36,7 @@
     "ipld-block": "^0.10.0"
   },
   "devDependencies": {
-    "aegir": "^26.0.0",
+    "aegir": "git://github.com/ipfs/aegir#eslint-config",
```
Comment (Gozala) [inline packages/ipfs-message-port-protocol/package.json:39]:
```suggestion
    "aegir": "git://github.com/ipfs/aegir#master",
```

File: packages/ipfs-message-port-server/package.json [hunk 6]
```diff
@@ -39,7 +39,7 @@
   },
   "devDependencies": {
     "@types/it-all": "^1.0.0",
-    "aegir": "^26.0.0",
+    "aegir": "git://github.com/ipfs/aegir#eslint-config",
```
Comment (Gozala) [inline packages/ipfs-message-port-server/package.json:42]:
```suggestion
    "aegir": "git://github.com/ipfs/aegir#master",
```

File: packages/ipfs/package.json [hunk 7]
```diff
@@ -172,7 +172,7 @@
     "yargs-promise": "^1.1.0"
   },
   "devDependencies": {
-    "aegir": "^26.0.0",
+    "aegir": "git://github.com/ipfs/aegir#eslint-config",
```
Comment (Gozala) [inline packages/ipfs/package.json:175]:
```suggestion
    "aegir": "git://github.com/ipfs/aegir#master",
```

**Comment by Gozala** [comment]
@jacobheun you've offered to take some reviews off of @achingbrain's plate. If you have time to review this it would be great to get this inn.

File: packages/interface-ipfs-core/package.json [hunk 8]
```diff
@@ -66,7 +66,7 @@
     "uint8arrays": "^1.1.0"
   },
   "devDependencies": {
-    "aegir": "^26.0.0",
+    "aegir": "git://github.com/ipfs/aegir#master",
```
Comment (achingbrain) [inline packages/interface-ipfs-core/package.json:69]: Can be changed to aegir@27.0.0 now in all the `package.json`s here

Comment (achingbrain) [reply packages/interface-ipfs-core/package.json]: No worries, I update them all.

**Review by achingbrain — APPROVED** [review]

**Pull request merged**